### ISSUE TITLE
Disable/hide irrelevant UI elements according to user permissions

### ISSUE THREAD

**Issue by sa-spag** [argoproj/argo-cd#10601]
# Summary

Disable/hide irrelevant elements on Argo CD UI according to user permissions configured through [RBAC](https://argo-cd.readthedocs.io/en/stable/operator-manual/rbac/).

# Motivation

Today, [as reported on Slack](https://cloud-native.slack.com/archives/C01TSERG0KZ/p1663255592122069), users with limited permissions can see all UI controls as if they could perform any action. For instance, users with read-only permissions on all resources can still see the sync, delete and other buttons as clickable on the application view. The UI eventually shows an error message when users attempt an actions they are not allowed to.

As a result, the UI is kind of overwhelming for some users.

# Proposal

Elements that correspond to disallowed actions could be disabled, e.g. greyed out, or even hidden.

**Comment by crenshaw-dev** [comment]
Here's an example of how this might be done: https://github.com/argoproj/argo-cd/pull/9828/files#diff-a5340e598cfacef4e52f66ac8f8a9174006c508fd12448cd4588a8de4c97e498R264

**Comment by lchastel** [comment]
I need these feature too.

In my case, I created a role "none" and used it as policy.default. So users can't do anything if they are not in the correct groups.
The issue is that the user could still see the "create" buttons for applications, repositories, ..., click on them.
It's only after filling the form and click on "save", that he received the message telling him he don't have the rights to perform the action.

**Comment by SaberStrat** [comment]
Am evaluating Argo CD right now and I was honestly surprised that the UI isn't role/RBAC dependent and that everybody sees every control element once they are authenticated.

**Comment by SaberStrat** [comment]
Similar to https://github.com/argoproj/argo-cd/issues/5013

**Comment by jan-mrm** [comment]
> Here's an example of how this might be done: https://github.com/argoproj/argo-cd/pull/9828/files#diff-a5340e598cfacef4e52f66ac8f8a9174006c508fd12448cd4588a8de4c97e498R264

this issue is somewhat old, but I stumbled across it. Think it would be cool but is that something that could be implemented? Three questions:
- would it be welcome?
- could it be done using "can-i"? Currently it seems that it can not check fine-grained permissions, or am I wrong? My current test example is a policy `p, u1, applications, delete/*/Pod/*/*, test-project/*, allow` but i didn't not figure out how to "can-i" that.   
`argocd account can-i delete applications 'test-project/*` is obviously false and correct.
But I'd imagine a command like `argocd account can-i delete applications 'test-project/*/*/Pod/*/*'` which does result in `no`. I didn't find any fine-grained example, so I am not sure it has that functionality yet (assuming the "can-i" from the cli is the same as the endpoint used in the ui from the PR you linked)
- should it be done using "can-i" (if its possible)? The more buttons or elements one would hide, the more "can-I" requests would be done

**Comment by agaudreault** [comment]
Closing as Duplicate of https://github.com/argoproj/argo-cd/issues/5013
